### ISSUE TITLE
uploadFile() for local file that doesn't exist, transferStatusUpdatedCallback isn't called

### ISSUE THREAD

**Issue by diablodale** [aws/aws-sdk-cpp#654]
When calling `uploadFile()` for local file that doesn't exist, the sdk never calls `transferStatusUpdatedCallback` for that transfer.

### Setup
Windows 10 x64
VS2017 15.3.3
AWS SDK 1.1.49

### Repro
When setting up transferManager and its transferConfig, create a lambda on the config's `transferStatusUpdatedCallback`. In that lambda, display some simple debug text.

Then with a valid transferManager, do
```cpp
transferManager->UploadFile("c:\\fileDoesNotExist.txt", "myBucket", "myKey", mimeType, Aws::Map<Aws::String, Aws::String>()));
```

### Result
The debug display you put in the lambda for `transferStatusUpdatedCallback` is never seen.

### Expected
Your debug display. `transferStatusUpdatedCallback` should be called by the sdk with

> transferStatus = FAILED
> GetTargetFilePath() = "c:\\fileDoesNotExist.txt"
> GetBucketName() = "myBucket"
> GetKey() = "myKey"
> GetTransferDirection() =TransferDirection::UPLOAD
> error = not_found
> etc.

**Comment by singku** [comment]
It's a bug. Currently transferStatusUpdatedCallback is only get called during the actual transfer.

**Comment by diablodale** [comment]
Hello. What is the resolution of this bug? If by code, what was the build? If other, please share with your customers the resolution.
For example, did TransferManager in 1.2 with commit abcdef resolve this by reporting an error on the callback errorCallback?

**Comment by singku** [comment]
Ah, it's been fixed in this commit: 55563eeecf809fc45630c67505025ffc4a624696

**Comment by wps132230** [comment]
We call the callback function if failed to read the file (status updated to `FAILED`)
https://github.com/aws/aws-sdk-cpp/blob/f5921a32c884928f45967b878fbff204a4075534/aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp#L953
And here is the related test in `TransferTest.cpp`
https://github.com/aws/aws-sdk-cpp/blob/f5921a32c884928f45967b878fbff204a4075534/aws-cpp-sdk-transfer-tests/TransferTests.cpp#L1362
This issue was closed automatically by Github because we referenced it in the commit, sorry for not updating immediately, and thank you for your suggestion again.
